src/generators/schema-generator.php

```diff
@@ -131,6 +145,23 @@ public function generate( Meta_Tags_Context $context ) {
 		];
 	}
 
+	/**
+	 * Retrieves the type from a graph piece.
+	 *
+	 * @param array $piece The graph piece.
+	 *
+	 * @return false|string False on failure, the piece's type on success.
+	 */
+	private function get_type_from_piece( $piece ) {
+		if ( isset( $piece['@type'] ) ) {
+			if ( is_array( $piece['@type'] ) ) {
+				return $piece['@type'][0];
```

Comment:
Would "more magic" be preferred here? Like imploding this with an "/" or "-"?
Not sure how specific the recipient of this filter needs this information to be.

Reply:
Not sure I understand what you mean...

Reply:
We are now only using the first item of the array; but for types that are an array, is this always sorted on most important first?

Reply:
Could also run it for each. Bit heavier, but more reliable.

Reply:
Reworked the code :)